### ISSUE TITLE
Lint/Void false positive and unsafe autocorrect

### ISSUE THREAD

**Issue by yayamochi** [rubocop/rubocop#12687]
Hi, I noticed a bug while using [Parallel)](https://github.com/grosser/parallel/tree/master) .

sample.rb
```ruby
# frozen_string_literal: true
require 'parallel'
items = [1, 2, 3]
Parallel.each(items,
              finish: lambda { |_item, _index, result|
                puts "puts result: #{result}"
              }) do |i|
  puts i * 10
  i * 10
end
```

## Expected behavior
```shell
bundle exec rubocop -a sample.rb
```
No changes should occur, and the following result should be obtained:

```
30
10
20
puts result: 10
puts result: 20
puts result: 30
```

## Actual behavior
The following error occurs:

```
Inspecting 1 file
W

Offenses:

sample.rb:11:3: W: [Corrected] Lint/Void: Variable i used in void context.
  i
  ^
sample.rb:11:5: W: [Corrected] Lint/Void: Operator * used in void context.
  i * 10
    ^
sample.rb:12:1: C: [Corrected] Layout/IndentationConsistency: Inconsistent indentation detected.
10
^^
sample.rb:12:1: W: [Corrected] Lint/Void: Literal 10 used in void context.
10
^^
sample.rb:12:3: W: [Corrected] Lint/Void: Literal 10 used in void context.
  10
  ^^

1 file inspected, 5 offenses detected, 5 offenses corrected
```

This code and results are changed.

```ruby
# frozen_string_literal: true
require 'parallel'
items = [1, 2, 3]
Parallel.each(items,
              finish: lambda { |_item, _index, result|
                puts "puts result: #{result}"
              }) do |i|
  puts i * 10
end
```

```
20
10
puts result:
puts result:
30
puts result:
```

I think `check_void_op` runs when 'each' is detected.
https://github.com/rubocop/rubocop/blob/master/lib/rubocop/cop/lint/void.rb#L86

Additionally, If you change the return value from `i * 10` to `(i * 10)`, nothing happens. which is storange.

## Steps to reproduce the problem
Test case above.
You can pull it from [this gist](https://gist.github.com/yayamochi/d9accbcf1c28e44274333d9725ce6273) and execute rubocop.

## RuboCop version

```
$ [bundle exec] rubocop -V
1.60.2 (using Parser 3.3.0.5, rubocop-ast 1.30.0, running on ruby 3.0.4) [arm64-darwin22]
```
